src/circuit/operator/inspect.rs

```diff
@@ -1,9 +1,41 @@
 //! Defines a sink operator that inspects every element of its input stream by applying a
 //! user-provided callback to it.
 
-use crate::circuit::operator_traits::{Operator, SinkOperator};
+use crate::circuit::{
+    operator_traits::{Operator, SinkOperator},
+    Circuit, Stream,
+};
 use std::{borrow::Cow, marker::PhantomData};
 
+impl<P, D> Stream<Circuit<P>, D>
+where
+    D: Clone + 'static,
```

Comment:
I think at this point it comes down to using `Cow` to wrap values read from streams (regardless of an operator). We don't truly need `Cow` as we never do actual copy-on-write, so we could define our own `enum` instead that wouldn't require `Clone`. It would have to be another PR though and doesn't seem very pressing, as we generally want `Clone` for types inside the dataflow.  But yeah, shouldn't be strictly necessary any longer.